WallsAndHoles/tilematerialselectiondialog.h

```diff
@@ -0,0 +1,23 @@
+#ifndef TILEMATERIALSELECTIONDIALOG_H
+#define TILEMATERIALSELECTIONDIALOG_H
+
+#include <QDialog>
+#include <QListView>
+
+class TileMaterialSelectionDialog : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit TileMaterialSelectionDialog(QWidget *parent = nullptr);
+
+    int result = 0; //default material
```

Comment:
It would be better to make this a private variable (`mResult`) and expose it through a getter (`int result()`)